File: hackertracker/src/main/java/com/shortstack/hackertracker/Renderer/VendorRenderer.java
package com.shortstack.hackertracker.Renderer;

import android.support.v7.app.AppCompatActivity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.pedrogomez.renderers.Renderer;
import com.shortstack.hackertracker.Fragment.VendorBottomSheetDialogFragment;
import com.shortstack.hackertracker.Model.Vendor;
import com.shortstack.hackertracker.R;

import java.util.List;

import butterknife.Bind;
import butterknife.ButterKnife;

public class VendorRenderer extends Renderer<Vendor> implements View.OnClickListener {

    @Bind(R.id.title)
    TextView title;

    @Bind(R.id.description)
    TextView description;

    @Override
    protected View inflate(LayoutInflater inflater, ViewGroup parent) {
        View view = inflater.inflate(R.layout.row_vendor, parent, false);
        ButterKnife.bind(this, view);
        return view;
    }

    @Override
    protected void hookListeners(View rootView) {
        super.hookListeners(rootView);
        rootView.setOnClickListener(this);
    }

    @Override
    public void render(List<Object> payloads) {
        title.setText(getContent().getTitle());
        description.setText(getContent().getDescription());
    }


    @Override
    public void onClick(View view) {
        VendorBottomSheetDialogFragment bottomSheetDialogFragment =  VendorBottomSheetDialogFragment.newInstance(getContent());
        bottomSheetDialogFragment.show(((AppCompatActivity)getContext()).getSupportFragmentManager(), bottomSheetDialogFragment.getTag());
    }
}

File: hackertracker/src/main/java/com/shortstack/hackertracker/Renderer/VendorRenderer.kt
package com.shortstack.hackertracker.Renderer

import android.support.v7.app.AppCompatActivity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.pedrogomez.renderers.Renderer
import com.shortstack.hackertracker.BottomSheet.VendorBottomSheetDialogFragment
import com.shortstack.hackertracker.Model.Vendors
import com.shortstack.hackertracker.R
import kotlinx.android.synthetic.main.row_vendor.view.*

class VendorRenderer : Renderer<Vendors.Vendor>(), View.OnClickListener {

    override fun inflate(inflater: LayoutInflater, parent: ViewGroup): View {
        return inflater.inflate(R.layout.row_vendor, parent, false)
    }

    override fun hookListeners(rootView: View?) {
        super.hookListeners(rootView)
        rootView!!.setOnClickListener(this)
    }

    override fun render(payloads: List<Any>) {
        rootView.title!!.text = content.title
        rootView.description!!.text = content.description
        rootView.partner!!.visibility = if (content.isPartner) View.VISIBLE else View.GONE
    }


    override fun onClick(view: View) {
        val bottomSheetDialogFragment = VendorBottomSheetDialogFragment.newInstance(content)
        bottomSheetDialogFragment.show((context as AppCompatActivity).supportFragmentManager, bottomSheetDialogFragment.tag)
    }
}
